### PR TITLE
Role Implementation

### PR DESCRIPTION
**Pull request by jan-kai1**
For #59 and #65

RoleHandler
RoleHandler.getRole changed to a static method. Since it accesses static fields which are constant across all RoleHandler instances and isValidRole is static, makes more sense getRole method should be static as well

Person
Person constructor takes in a varargs of Role ... (Unsure if should make roles mandatory)
Person Constructor takes in a Set

Person instance now has a Set along with methods getRoles() and hasRole
In line with existing edit command process (which creates a new Person instance and replaces the existing one in addressBook), Person currently does NOT have any methods to set or change the existing roles.
This can be changed based on later implementation of addRole command.

Add and Edit Commands
Now supports addition of roles, currently have to input each role individually e.g. /r vendor /r attendee

JsonAdaptedRole
Takes in any string with no checks, exception only thrown when toModelType() is called

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -102,11 +105,17 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
-        TelegramUsername telegramUsername = editPersonDescriptor.getTelegramUsername()
+        TelegramUsername updatedTeleUsername = editPersonDescriptor.getTelegramUsername()
                 .orElse(personToEdit.getTelegramUsername());
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
 
-        return new Person(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedTags, telegramUsername);
+        //Role[] updatedRoles = (editPersonDescriptor.getRoles().orElse(personToEdit.getRoles())).toArray(new Role[0]);
```
Comment (cshao02) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:112]: Maybe this line can be removed if it is no longer used

File: src/main/java/seedu/address/model/role/Role.java [hunk 2]
```diff
@@ -16,12 +16,15 @@
...
 public abstract class Role {
-    public static final String MESSAGE_CONSTRAINTS = "Roles should be alphanumeric";
+    public static final String MESSAGE_CONSTRAINTS = "Roles should be one of the following:"
+            + "\n" + Attendee.ROLE_WORD
+            + "\n" + Sponsor.ROLE_WORD
+            + "\n" + Vendor.ROLE_WORD
+            + "\n" + Volunteer.ROLE_WORD;
     public static final String VALIDATION_REGEX = "\\p{Alnum}+";
 
     private final HashSet<Person> personList = new HashSet<>();
```
Comment (cshao02) [inline src/main/java/seedu/address/model/role/Role.java:26]: maybe persons may be better cuz it's not rly a list?

**Review by cshao02 — APPROVED** [review]
LGTM! Maybe just some minor refinements will be good, such as removing commented out code if it's no longer used

**Comment by jan-kai1** [comment]
Closed, updated with #80

**Pull request closed without merging**